packages/usace/usace/lib/locations_collection.py

```diff
@@ -1,35 +1,210 @@
 # Copyright 2025 Lincoln Institute of Land Policy
 # SPDX-License-Identifier: MIT
 
-import json
+from typing import Optional, cast, assert_never
 from com.cache import RedisCache
-from com.geojson.helpers import GeojsonFeatureCollectionDict, GeojsonFeatureDict
-from com.helpers import await_
-import msgspec
-from usace.lib import types
+from com.env import TRACER
+from com.geojson.helpers import (
+    GeojsonFeatureCollectionDict,
+    GeojsonFeatureDict,
+    SortDict,
+    all_properties_found_in_feature,
+    filter_out_properties_not_selected,
+    sort_by_properties_in_place,
+)
+from com.helpers import (
+    EDRFieldsMapping,
+    OAFFieldsMapping,
+    await_,
+    parse_bbox,
+    parse_z,
+)
+import geojson_pydantic
+import orjson
+from rise.lib.covjson.types import CoverageCollectionDict
+from rise.lib.types.helpers import ZType
+import shapely
+from usace.lib.types.geojson_response import FeatureCollection
+import shapely.wkt
+from geojson_pydantic.types import Position2D
 
 
-class LocationColletion:
+class LocationCollection:
     def __init__(self):
         self.cache = RedisCache()
         url = "https://water.sec.usace.army.mil/cda/reporting/providers/projects?fmt=geojson"
 
         res = await_(self.cache.get_or_fetch_response_text(url))
-        assert msgspec.json.decode(res, type=list[types.Project]), (
-            f"{res} did not match expected schema"
+        self.fc = FeatureCollection.model_validate(
+            {
+                "type": "FeatureCollection",
+                "features": orjson.loads(res),
+            }
         )
-        self.locations: list[GeojsonFeatureDict] = json.loads(res)
-        for loc in self.locations:
-            loc["id"] = loc["properties"]["location_code"]
-            loc["properties"]["name"] = loc["properties"]["public_name"]
+        for loc in self.fc.features:
+            loc.id = str(loc.properties.location_code)
+            loc.properties.name = loc.properties.public_name
```

Comment:
Do we need to iterate through all features here? Seems like it could be done in `to_geojson`?